woocommerce/woocommerce
# [testing workflows] Add workflow_call trigger in CI workflow - support daily and other types of runs

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).

### Changes proposed in this Pull Request:

Prep work for #47532 

Adds the `workflow_call` trigger for `ci.yml`. This makes it possible to trigger the jobs defined in `ci.yml` by other events.

Instead of defining all the possible triggers in ci.yml directly, I think it would be cleaner to have other workflows that are triggered by those events (e.g. cron) and have those workflows call `ci.yml` with specific inputs (e.g. `daily-e2e`, `release-checks`, etc.). 

This PR adds the `tests-daily-run.yml` workflow which will run on a schedule and call `ci.yml` with the `daily-e2e` input. The `ci-jobs` utility will return a jobs matrix with only those jobs configured to run on `daily-e2e` event - currently only `Core e2e tests - Gutenberg`.

I only configured the Gutenberg job as an example and to test it. All other daily or release test projects will be configured in follow-up PRs and `smoke-test-daily.yml` and 
`smoke-test-release.yml` will be deprecated.

I had to update the events for all test jobs in all projects files so that they will not run on daily runs. I did not update the lint jobs because they are restricted at workflow level since #47652.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

- Tested with [this run](https://github.com/woocommerce/woocommerce/actions/runs/9190932987), where ci.yml was triggered with a `workflow_call` event by `tests-daily-run.yml` with `inputs.trigger=daily-e2e`. Only the e2e tests with Gutenberg ran (ignore the lint jobs, they will not run once #47652 is merged). The tests are expected to fail, you can ignore the results.
- Check code changes.
- CI jobs run as before.
